### PR TITLE
[python] Avoid context leak on group-set-timestamp operation; use TileDB Embedded 2.15.1

### PR DESCRIPTION
**Pull request by johnkerl**
**Issue and/or context:** #1169. See also a follow-on tracker which is #1198.

See also #1222

### CODE REVIEW

File: apis/python/src/tiledbsoma/_tdb_handles.py [hunk 1]
```diff
@@ -220,9 +220,24 @@ def _opener(
...
+        # * Timestamps are read from tiledb config.
+        # * We can't set-config on an already opened group.
+        # * We could create a new ctx, with the current ctx's config plus the
+        #   new timestamp config, but that causes a thread-pool leak. See also
+        #   https://github.com/single-cell-data/TileDB-SOMA/issues/1169.
+        # * For now, we do this close, set-config, and re-open workaround.
+        # * An upcoming TileDB-Py release will remove this need for us.
+        # TODO: make that happen.
+        # Tracking issue: https://github.com/single-cell-data/TileDB-SOMA/issues/1198
+        G = tiledb.Group(uri, mode)
```
Comment (Shelnutt2) [inline apis/python/src/tiledbsoma/_tdb_handles.py:233]: I suggest we wait for the next TileDB-Py version where we just take the config on start, and avoid having to open and close.

Comment (nguyenv) [reply apis/python/src/tiledbsoma/_tdb_handles.py:233]: FYI TileDB-Py 0.21.2 was released this morning and includes the addition of a `Group` constructor that takes in a `Config` argument.

Comment (johnkerl) [reply apis/python/src/tiledbsoma/_tdb_handles.py:233]: @nguyenv on it 😎

**Comment by johnkerl** [comment]
This is blocked until TileDB-Py 0.21.2 reaches PyPI:
https://pypi.org/project/tiledb/#history

**Comment by johnkerl** [comment]
@nguyenv good to go?

**Review by nguyenv — APPROVED** [review]
Yes looks good to me.

**Pull request merged**